### PR TITLE
Remove the check for large notification index.

### PR DESCRIPTION
**Pull request by msramek**
In shouldShowNotification_, we immediately return true for large indices. This is the correct behavior, since it allows us to try again if the extension uninstallation failed.

However, in maybeShowNotification_, we set the index to zero if it is out of range, which means that we reset the timeline. This CL removes the upper bound check in maybeShowNotification_, so that this doesn't happen.

To test that the whole process works, we add a mock of Chrome extension APIs.

### CODE REVIEW

File: chrome/test/chrome-api-shim.js [hunk 1]
```diff
@@ -1 +1,60 @@
-// Shim various `chrome.*` APIs.
+var chrome = {};
+
+(function() {
+  /* chrome.storage */
+  var storage_ = {};
+
+  function copy_dict(from, to) {
+    for (item in from)
```
Comment (mikewest) [inline chrome/test/chrome-api-shim.js]: Nit: {} for multi-line clauses.

File: chrome/test/sunset-test.js [hunk 2]
```diff
@@ -45,3 +45,40 @@ test(function () {
   current.setSeconds(current.getSeconds() - 1);
   assert_false(Sunset.shouldShowNotification_(start, current, 1));
 }, "Edge-cases for shouldShowNotification_");
+
+test(function () {
```
Comment (mikewest) [inline chrome/test/sunset-test.js]: Nit: This really ought to be an `async_test` with `step_func`s for each callback, but I suppose it makes no difference given how you've structured it (and given that the callbacks aren't really async in the mock).

**Comment by mikewest** [comment]
LGTM % nits. Land it whenever you're ready.

File: chrome/test/sunset-test.js (hunk 2 above)
Comment (msramek) [inline chrome/test/sunset-test.js]: Right, I didn't notice that one. And yes, if the mock wasn't asynchronous, maybeShowNotification_() in the test wouldn't be asynchronous, and we couldn't have immediately called chrome.storage.sync.get. Nevertheless, it works well right now - I verified that this test has the good property that any small change in the code or in itself turns it red :)

File: chrome/test/chrome-api-shim.js (hunk 1 above)
Comment (msramek) [inline chrome/test/chrome-api-shim.js]: Done.

**Comment by msramek** [comment]
Thanks! Landing.
